src/main/java/com/parasoft/demoapp/config/activemq/RefreshableMessageListener.java

```diff
@@ -1,28 +1,47 @@
 package com.parasoft.demoapp.config.activemq;
 
+import com.parasoft.demoapp.exception.GlobalPreferencesMoreThanOneException;
+import com.parasoft.demoapp.exception.GlobalPreferencesNotFoundException;
+import com.parasoft.demoapp.service.GlobalPreferencesDefaultSettingsService;
+import com.parasoft.demoapp.service.GlobalPreferencesService;
 import org.springframework.jms.config.DefaultJmsListenerContainerFactory;
 import org.springframework.jms.config.JmsListenerEndpointRegistry;
 import org.springframework.jms.config.SimpleJmsListenerEndpoint;
+import org.springframework.jms.connection.CachingConnectionFactory;
+import org.springframework.jms.listener.MessageListenerContainer;
 import org.springframework.jms.support.converter.MessageConverter;
 
 import javax.jms.Message;
 import javax.jms.MessageListener;
+import java.util.HashMap;
+import java.util.Map;
 
 public abstract class RefreshableMessageListener implements MessageListener {
 
     private static final String ID_PREFIX = "id:";
 
+    protected final CachingConnectionFactory cachingConnectionFactory;
+
     protected final MessageConverter jmsMessageConverter;
     private final JmsListenerEndpointRegistry jmsListenerEndpointRegistry;
     private final DefaultJmsListenerContainerFactory jmsQueueListenerContainerFactory;
 
+    private final GlobalPreferencesService globalPreferencesService;
+
+    private final GlobalPreferencesDefaultSettingsService defaultGlobalPreferencesSettingsService;
```

Comment:
defaultGlobalPreferencesSettingsService property is not used in this class, remove it.